05_ANALYSIS_TOOLS/etl_pipeline/extract/financial_extractor.py

```
@@ -0,0 +1,215 @@
+                    for encoding in ['utf-8', 'latin-1', 'cp1252', 'iso-8859-1']:
+                        try:
+                            df = pd.read_csv(file_path, encoding=encoding, low_memory=False)
+                            filename = Path(file_path).stem
+                            pnl_data[f'pnl_2023_{filename}'] = df
+                            logger.info(f"Loaded P&L 2023 data: {filename} ({len(df)} records) with {encoding} encoding")
+                            break
+                        except UnicodeDecodeError:
+                            continue
+                    else:
+                        logger.error(f"Could not decode P&L 2023 file {file_path} with any encoding")
+                except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)